extensions/elytron-security/runtime/src/main/java/io/quarkus/elytron/security/runtime/ElytronRecorder.java

```diff
@@ -80,8 +80,9 @@ public boolean implies(Permission permission) {
      * @param realmName - the name of the SecurityRealm
      * @param realm - the runtime value for the SecurityRealm
      */
-    public void addRealm(RuntimeValue<SecurityDomain.Builder> builder, String realmName, RuntimeValue<SecurityRealm> realm) {
-        builder.getValue().addRealm(realmName, realm.getValue());
+    public RuntimeValue<SecurityDomain.Builder> addRealm(RuntimeValue<SecurityDomain.Builder> builder, String realmName,
+            RuntimeValue<SecurityRealm> realm) {
+        return new RuntimeValue<>(builder.getValue().addRealm(realmName, realm.getValue()).build());
```

Comment:
I agree the build is missing but do you really need to return the builder? It seems useless to me but I might miss something.

Reply:
Yep, it is indeed not needed, I will remove it. The `build` call is enough

Reply:
Cool, thanks. I squashed the two commits and force pushed. Let's wait for CI!